### PR TITLE
Hifio Cassandra and  Elasticsearch Part2 fixes

### PR DESCRIPTION
**Pull request by diptikul**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by ssisk** [comment]
hey - while you've removed all the old commits, you still have all the old changes to the files - if you look in the "files" tab of this PR, you can see that you're changing a lot of files that aren't what I think you intend to change.

**Comment by diptikul** [comment]
@ssisk  Please review EmbeddedCassandraTest and populateDisplayData movement in this as you have already reviewed the getFractionsConsumedTest in the other PR #2423.  For your comment on the test of getFractionsConsumed, it will be addressed on the other PR #2423

File: sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java [hunk 1]
```diff
@@ -472,7 +478,8 @@ public void testReadWithNullCreateRecordReader() throws Exception {
    */
   @Test
   public void testReadersStartWhenZeroRecords() throws Exception {
-    InputFormat<Text, Employee> mockInputFormat = Mockito.mock(EmployeeInputFormat.class);
+
```
Comment (ssisk) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: non-blocking check-in: extra space.

File: sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java [hunk 2]
```diff
@@ -546,6 +554,33 @@ public void testReadersGetFractionConsumed() throws Exception {
...
+            serConf,
+            WritableCoder.of(Text.class),
+            AvroCoder.of(Employee.class),
+            null, // No key translation required.
+            null, // No value translation required.
+            new SerializableSplit(mockInputSplit));
+    boundedSource.setInputFormatObj(mockInputFormat);
+    BoundedReader<KV<Text, Employee>> boundedReader = boundedSource.createReader(p.getOptions());
+    assertEquals(true, boundedReader.start());
+    assertEquals(null, boundedReader.getFractionConsumed());
```
Comment (ssisk) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: can we do a second read from this source to make sure getFractionConsumed works after advance() is called?

Comment (ssisk) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: basically - what happens in the scenario where you're reading records just fine and then a bad progress value is returned?

File: sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    dropKeyspaces();
+    session.close();
+    cluster.close();
+  }
+
+  /**
+   * Drop all keyspaces (expect system).
+   */
+  private static void dropKeyspaces() {
+    List<String> keyspaces = new ArrayList<String>();
```
Comment (ssisk) [inline sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java]: if we destroying the cluster, is this important to do?

File: sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+   * successfully.
+   * @throws Exception
+   */
+  @Test
+  public void testHIFReadForCassandra() throws Exception {
+    // Expected hashcode is evaluated during insertion time one time and hardcoded here.
+    String expectedHashCode = "4651110ba1ef2cd3a7315091ca27877b18fceb0e";
+    Configuration conf = getConfiguration();
+    SimpleFunction<Row, String> myValueTranslate = new SimpleFunction<Row, String>() {
+      @Override
```
Comment (ssisk) [inline sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java]: please share this code between the two versions

File: sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java (hunk 3 above)
Comment (diptikul) [inline sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java]: Not really, will remove it.

File: sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java (hunk 4 above)
Comment (diptikul) [inline sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HIFIOWithEmbeddedCassandraTest.java]: Done

File: sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java (hunk 2 above)
Comment (diptikul) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: Added a second read. while the read is happening and progress value is returned as 2.0 which is bad value, the getProgress will return null.

File: sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java (hunk 1 above)
Comment (diptikul) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: Removed

Comment (ssisk) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: I don't see it fixed in the most recent version of the file - https://github.com/apache/beam/pull/2460/files#diff-aa7ab5dd37b6540d3c9f8b13ed4fd7f7L475

**Comment by ssisk** [comment]
R @dhalperi - mind merging this?

**Comment by dhalperi** [comment]
Merged on @ssisk 's LGTM :)

File: sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java (hunk 1 above)
Comment (diptikul) [inline sdks/java/io/hadoop/input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIOTest.java]: Removed, oops ill add it again

**Pull request closed without merging**